lib/engine/route.rb

```diff
@@ -220,7 +220,7 @@ def revenue
         @game.game_error("Cannot use group #{key} more than once") unless group.one?
       end
 
-      @game.revenue_for(self, stops)
+      @game.revenue_for(self, stops, routes: @routes)
```

Comment:
You don't need to pass @routes, since you are already passing self. the revenue_for method can already access routes via that reference.